example/storybook/index.tsx

```diff
@@ -32,6 +32,7 @@ configure(() => {
 // To find allowed options for getStorybookUI
 const StorybookUIRoot = getStorybookUI({
   asyncStorage: null,
+  onDeviceUI: false,
 });
```

Comment:
This is the main change of this PR.  If you agree, I'd like to turn off the device the on-simulator storybook UI (and thus rely on `yarn web-controls` to navigate the stories.  This is because the storybook UI automatically adds margin at the top and bottom of the screen and makes debugging things like the header and tab navigation nearly impossible.

Here are screenshots of the before and after of this change alone:

| Before | After |
|-|-|
| ![Screenshot 2023-05-31 at 8 36 22 AM](https://github.com/lifeomic/react-native-sdk/assets/220893/3ca52d84-0b45-4215-8633-0fb17d90a797) | ![Screenshot 2023-05-31 at 8 36 08 AM](https://github.com/lifeomic/react-native-sdk/assets/220893/a061e367-d90b-4efb-84ec-4fa0eaa1666f) |


Obviously that isn't ideal. I designed the header and tab navigation not realizing that that auto margin was there.  

The rest of this PR fixes the stories by adding safe-area-view to individual components and fixes the header issue on the full Example app stories.

The tab navigation will be fixed in the next PR